### PR TITLE
[SH4RE-77] - Calendar Showing Disabled Days

### PR DESCRIPTION
**Pull request by hannahbrooks**
Video demo: https://streamable.com/yvi2wg

Changes:
- Moved to a new calendar library
- Allows for disabled days
- Not sure how to change the background colour

### CODE REVIEW

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 1]
```diff
@@ -169,6 +169,8 @@ struct ViewListingView: View {
         .padding([.horizontal])
         .frame(alignment: .bottom)
         .onAppear() {
+            myRkManager.disabledDates = [Date().addingTimeInterval(60*60*24*4), Date().addingTimeInterval(60*60*24*5), Date().addingTimeInterval(60*60*24*6)] // here is where we would add the disabled dates
+
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: disabled days

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 2]
```diff
@@ -185,7 +187,9 @@ struct ViewListingView: View {
                 }
             }
         }
+        .sheet(isPresented: $showCal) {
+            RKViewController(isPresented: $showCal, rkManager: myRkManager)
+        }
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: the sheet

File: SH4RE/SH4RE/Components/Calendar/RKColorSettings.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    @Published var textBackColor: Color = Color.white
+    @Published var todayBackColor: Color = Color.gray
+    @Published var selectedBackColor: Color = Color.primaryLight
+    @Published var disabledBackColor: Color = Color.clear
+    @Published var betweenStartAndEndBackColor: Color = Color.blue
+    // headers foreground colors
+    @Published var weekdayHeaderColor: Color = Color.primary
+    @Published var monthHeaderColor: Color = Color.primary
+    // headers background colors
+    @Published var weekdayHeaderBackColor: Color = Color.clear
```
Comment (americobarros) [inline SH4RE/SH4RE/Components/Calendar/RKColorSettings.swift:30]: Would this be there you change BG colour? Idk i am not terribly worried. I think the UI is not great of this library, but it works and maybe that is okay for now.

File: SH4RE/SH4RE/Components/Calendar/RKViewController.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+struct RKViewController: View {
+    
+    @Binding var isPresented: Bool
+    
+    @ObservedObject var rkManager: RKManager
+    
+    var body: some View {
+        Group {
+//            RKWeekdayHeader(rkManager: self.rkManager)
```
Comment (americobarros) [inline SH4RE/SH4RE/Components/Calendar/RKViewController.swift:19]: why this commented out? what is?

File: SH4RE/SH4RE/Components/Calendar/RKViewController.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        
+        return rkManager.calendar.date(from: components)!
+    }
+}
+
+#if DEBUG
+struct RKViewController_Previews : PreviewProvider {
+    static var previews: some View {
+        Group {
+//            RKViewController(isPresented: .constant(false), rkManager: RKManager(calendar: Calendar.current, minimumDate: Date(), maximumDate: Date().addingTimeInterval(60*60*24*365), mode: 0))
```
Comment (americobarros) [inline SH4RE/SH4RE/Components/Calendar/RKViewController.swift]: why remove from preview?

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 6]
```diff
@@ -35,6 +35,9 @@ struct ViewListingView: View {
     @State private var dates: Set<DateComponents> = []
     
     
+    var myRkManager = RKManager(calendar: Calendar.current, minimumDate: Date(), maximumDate: Date().addingTimeInterval(60*60*24*365), mode: 3)
```
Comment (americobarros) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: not a huuuuge fan of this var name... RK is the initials of the dev, like we GET IT, you made the lib...
Whatever, nbd tho.

**Review by americobarros — CHANGES_REQUESTED** [review]
Did you take a look at implementing something like [this](https://nemecek.be/blog/161/how-to-use-uicalendarview-in-ios) instead, so we could have avoided the use of libraries and kept the old look?
(Also, theres a bug rn where the disabled dates get cleared when you click onto a different page then back.)[https://streamable.com/trraz0] Thats the only thing I think would need to be fixed.

File: SH4RE/SH4RE/Components/Calendar/RKViewController.swift (hunk 4 above)
Comment (hannahbrooks) [reply SH4RE/SH4RE/Components/Calendar/RKViewController.swift:19]: part of the orig design i didnt like

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift (hunk 6 above)
Comment (hannahbrooks) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift]: ya i can change

File: SH4RE/SH4RE/Components/Calendar/RKColorSettings.swift (hunk 3 above)
Comment (hannahbrooks) [reply SH4RE/SH4RE/Components/Calendar/RKColorSettings.swift:30]: ya i fiddled w it a bunch but just decided to leave it

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 7]
```diff
@@ -89,24 +92,11 @@ struct ViewListingView: View {
...
-            //                        }
-            //                        .frame(alignment: .trailing)
-            //                        .padding()
-            //                        .background(Color.primaryDark)
-            //                        .cornerRadius(40)
-            //                        .padding()
             
             Button(action: {
-                showPopUp.toggle()
+//                if (startDateText != "") {
```
Comment (WasifButt) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: can we remove comments

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 8]
```diff
@@ -118,10 +108,12 @@ struct ViewListingView: View {
                 }
                 .frame(alignment: .trailing)
                 .padding()
-                .background(Color.primaryDark)
+                .background(startDateText == "" ? Color.grey : Color.primaryDark)
```
Comment (WasifButt) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift:109]: can we use .gray not .grey

**Review by WasifButt — CHANGES_REQUESTED** [review]

**Review by americobarros — APPROVED** [review]

**Review by WasifButt — APPROVED** [review]

**Pull request merged**